Sources/MongoSwift/SDAM.swift

```diff
@@ -3,12 +3,21 @@ import Foundation
 
 /// A struct representing a network address, consisting of a host and port.
 public struct ServerAddress: Equatable {
+    public enum HostType: String {
+        case ipv4
+        case ipLiteral = "ip_literal"
+        case hostname
+        case unixDomainSocket
+    }
+
     /// The hostname or IP address.
     public let host: String
 
     /// The port number.
     public let port: UInt16
 
+    private var type: HostType = .hostname
```

Comment:
Rather than having this be mutable with a default type, let's have it be immutable and just set it in the initializers.